### PR TITLE
Add clippy and resolve suggestions

### PR DESCRIPTION
**Pull request by nickfarrow**
Small clippy fixes and also removed the pointless `sid_len` from the `derive_nonce_rng!` that doesn't add any collision resistance. Though this will **break consistency of deterministic nonce rngs between releases**.

Prior to this fixup, the `&&`s have just been getting fully dereferenced right?
Like
```
Scalar::from_hash(hash.add(R).add(X).add(&&Message))
```

### CODE REVIEW

File: schnorr_fun/src/frost.rs [hunk 1]
```diff
@@ -493,14 +491,13 @@ impl<H: Digest<OutputSize = U32> + Clone, NG: NonceGen> Frost<H, NG> {
...
     ) -> R {
-        let sid_len = (session_id.len() as u64).to_be_bytes();
         let threshold_bytes = (frost_key.threshold() as u64).to_be_bytes();
         let pk_bytes = frost_key.public_key().to_xonly_bytes();
 
         let rng: R = derive_nonce_rng!(
             nonce_gen => self.nonce_gen(),
             secret => &secret,
-            public => [pk_bytes, threshold_bytes, sid_len, session_id],
+            public => [pk_bytes, threshold_bytes, session_id],
```
Comment (LLFourn) [inline schnorr_fun/src/frost.rs:500]: I'd prefer to keep this. Anything that is variable length should be length prefixed. I actually want to do this explicitly in the hashInto implementation.

**Comment by LLFourn** [comment]
Cool. Thanks. can you add clippy to CI so we can finally close #32.

**Comment by nickfarrow** [comment]
I noticed only a few uppercase acronyms (https://github.com/LLFourn/secp256kfun/issues/31)
```
type DLEQ = Eq<$mod::DLG<$len>, $mod::DL<$len>>;
```
Currently i have left them as clippy allowed

File: schnorr_fun/src/message.rs [hunk 2]
```diff
@@ -37,6 +37,11 @@ impl<'a, 'b, S: Secrecy> Message<'a, S> {
         }
     }
 
+    /// Check if the message is empty with zero length
+    pub fn is_empty(&self) -> bool {
```
Comment (LLFourn) [inline schnorr_fun/src/message.rs:41]: Did clippy make you do this?

**Comment by LLFourn** [comment]
Thanks so much for this!

File: schnorr_fun/src/message.rs (hunk 2 above)
Comment (nickfarrow) [reply schnorr_fun/src/message.rs:41]: Yeah, something about implementing len but not empty

**Pull request merged**